### PR TITLE
Implement React Flow sequence designer

### PR DESCRIPTION
**Pull request by sfmskywalker**
## Summary
- add SpecKit artifacts for the Sequence activity visual designer
- add a React Flow-backed Sequence designer, mapper, provider, and wrapper
- constrain Sequence mode to ordered structural edges with vertical/horizontal layout controls
- add focused Sequence mapper tests for order, orientation, invalid-edge discard, and embedded-region preservation

## Verification
- dotnet build Elsa.Studio.sln
- dotnet test Elsa.Studio.sln
- dotnet test src/modules/Elsa.Studio.Workflows.Designer.Tests/Elsa.Studio.Workflows.Designer.Tests.csproj --no-restore
- npm run build (src/modules/Elsa.Studio.Workflows.Designer/ClientLib)

## Notes
- Manual browser verification against a running Studio host is still recommended for large Sequences and Flowchart regression checks.

### CODE REVIEW

**Comment by sfmskywalker** [comment]
Addressed Copilot feedback in f8e6dba5:

- Removed the unused activity lookup from SequenceFlowDesigner.
- Fixed Sequence reload handling so a new JsonObject instance with the same node id still refreshes the graph.
- Multi-targeted the mapper tests across net8.0, net9.0, and net10.0.
- Replaced generic MapperFactory XML docs with specific summaries.

I did not wire React toolbar localization in this PR because the existing React designer client currently does not have a .NET localization payload/bridge for client-only toolbar strings. The Sequence toolbar follows the current React-side pattern; adding a shared localization contract should be a follow-up across the React designer surface rather than a Sequence-only partial fix.

**Comment by sfmskywalker** [comment]
Addressed the latest Copilot re-review comment in 7a519442:

- Removed the direct `ActivitySelected.InvokeAsync(Sequence)` call from `SequenceFlowDesigner.HandleCanvasSelected`.
- Canvas selection now flows through `CanvasSelected` only, with `SequenceDesignerWrapper.OnCanvasSelected` remaining the single source that selects the Sequence root.

Validation after the fix:
- `dotnet test src/modules/Elsa.Studio.Workflows.Designer.Tests/Elsa.Studio.Workflows.Designer.Tests.csproj --no-restore`
- `npm run build` in `src/modules/Elsa.Studio.Workflows.Designer/ClientLib`

**Comment by sfmskywalker** [comment]
Addressed the latest code-level Copilot feedback in 66b71f66:

- `SequenceMapper` now persists `sequenceLayoutOrientation` only for non-default horizontal layout and clears orientation metadata when returning to default vertical, avoiding legacy metadata churn.
- `SequenceFlowDesigner.OnParametersSetAsync` now invokes the debounced reload once per parameter-set cycle.
- Pasted activity ID regeneration now uses a shared helper for top-level and embedded activities, making the recursion path explicit.
- Added mapper tests covering no default orientation metadata and clearing orientation metadata when returning to vertical.

Validation:
- `dotnet test src/modules/Elsa.Studio.Workflows.Designer.Tests/Elsa.Studio.Workflows.Designer.Tests.csproj --no-restore` passed on net8.0, net9.0, and net10.0.

The repeated React toolbar localization comment remains intentionally deferred: the existing React designer does not have a .NET-to-React localization payload/bridge for client-only toolbar strings, so this should be handled as a shared React designer localization follow-up rather than a Sequence-only partial mechanism.

**Comment by sfmskywalker** [comment]
@greptile please review

File: src/modules/Elsa.Studio.Workflows.Designer/ClientLib/src/react-designer/components/Designer.tsx [hunk 1]
```diff
@@ -209,6 +211,66 @@ function fromReactFlowNode(node: Node<ActivityNodeData>): ElsaActivityNode {
...
+): Node<ActivityNodeData>[] {
+    const ordered = sortSequenceNodes(nodes, orientation);
+    const index = ordered.findIndex(n => n.selected);
+    if (index < 0) return nodes;
+    const target = index + direction;
+    if (target < 0 || target >= ordered.length) return nodes;
+    const next = [...ordered];
+    [next[index], next[target]] = [next[target], next[index]];
+    return arrangeSequenceNodes(next, orientation);
+}
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Studio.Workflows.Designer/ClientLib/src/react-designer/components/Designer.tsx:276]: Fixed in 1a549c7e. `moveSequenceSelection` now assigns positions from the swapped order via `arrangeSequenceNodesInOrder`, so Earlier/Later no longer gets undone by re-sorting old positions.

File: src/modules/Elsa.Studio.Workflows.Designer/Components/SequenceFlowDesigner.razor.cs [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+        foreach (var activity in activities)
+        {
+            var descriptor = ActivityRegistry.Find(activity.GetTypeName(), activity.GetVersion())!;
+            var newActivityId = IdentityGenerator.GenerateId();
+
+            activityLookup[activity.GetId()] = activity;
+            activity.SetId(newActivityId);
+            activity.SetNodeId($"{container.GetNodeId()}:{newActivityId}");
+            ProcessEmbeddedPorts(activity, descriptor);
+        }
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Studio.Workflows.Designer/Components/SequenceFlowDesigner.razor.cs]: Fixed in f8e6dba5. Removed the unused `activityLookup` dictionary from `SequenceFlowDesigner` since Sequence does not persist graph connections.

File: src/modules/Elsa.Studio.Workflows.Designer/ClientLib/src/react-designer/components/Designer.tsx [hunk 3]
```diff
@@ -1074,6 +1240,46 @@ const InnerDesigner = forwardRef<DesignerHandle, DesignerProps>(function InnerDe
...
+                                    type="button"
+                                    className="elsa-react-flow-toolbar-btn"
+                                    onClick={() => moveSelectedSequenceNode(1)}
+                                    title="Move selected activity later"
+                                    aria-label="Move selected activity later"
+                                >
+                                    Later
+                                </button>
+                            </>
+                        )}
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Studio.Workflows.Designer/ClientLib/src/react-designer/components/Designer.tsx:1286]: No code change in this PR. The existing React designer does not currently have a shared .NET-to-React localization payload for client-only toolbar strings, so a proper fix should be a React designer-wide localization follow-up rather than a Sequence-only partial mechanism.

File: src/modules/Elsa.Studio.Workflows.Designer.Tests/Elsa.Studio.Workflows.Designer.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFrameworks>net10.0</TargetFrameworks>
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Studio.Workflows.Designer.Tests/Elsa.Studio.Workflows.Designer.Tests.csproj]: Fixed in f8e6dba5. The mapper test project now targets net8.0, net9.0, and net10.0 to match the shipped designer TFMs.

File: src/modules/Elsa.Studio.Workflows.Designer/Services/MapperFactory.cs [hunk 5]
```diff
@@ -15,6 +15,15 @@ public async Task<IFlowchartMapper> CreateFlowchartMapperAsync(CancellationToken
         return new FlowchartMapper(activityMapper);
     }
 
+    /// <summary>
+    /// Provides the task.
+    /// </summary>
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Studio.Workflows.Designer/Services/MapperFactory.cs:20]: Fixed in f8e6dba5. Replaced the generic MapperFactory XML summaries with specific descriptions for flowchart, sequence, and activity mapper creation.

File: src/modules/Elsa.Studio.Workflows.Designer/Components/SequenceFlowDesigner.razor.cs [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+            _sequence = Sequence;
+            await _rateLimitedLoadSequenceAction.InvokeAsync();
+        }
+
+        if (!Equals(_activityStats, ActivityStats))
+        {
+            _activityStats = ActivityStats;
+            await _rateLimitedLoadSequenceAction.InvokeAsync();
+        }
+    }
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Studio.Workflows.Designer/Components/SequenceFlowDesigner.razor.cs:266]: Fixed in f8e6dba5. `SequenceFlowDesigner` now reloads when the Sequence instance changes even if the node id is the same, and the wrapper only skips reload when both id and object reference are unchanged.

File: src/modules/Elsa.Studio.Workflows.Designer/Components/SequenceFlowDesigner.razor.cs [hunk 7]
```diff
@@ -0,0 +1,331 @@
...
+        if (ActivityEmbeddedPortSelected.HasDelegate)
+            await ActivityEmbeddedPortSelected.InvokeAsync(new ActivityEmbeddedPortSelectedArgs(activity, portName));
+    }
+
+    [JSInvokable]
+    public async Task HandleCanvasSelected()
+    {
+        if (ActivitySelected.HasDelegate)
+            await ActivitySelected.InvokeAsync(Sequence);
+
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Studio.Workflows.Designer/Components/SequenceFlowDesigner.razor.cs]: Fixed in 7a519442. `HandleCanvasSelected` now raises only `CanvasSelected`; the wrapper remains the single place that selects the Sequence root.

File: src/modules/Elsa.Studio.Workflows.Designer/Components/SequenceFlowDesigner.razor.cs [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+            activity.SetNodeId($"{container.GetNodeId()}:{newActivityId}");
+            activity.SetName(newName);
+            activityNode.Id = newActivityId;
+
+            allActivities.Add(activity);
+            ProcessEmbeddedPorts(activity, descriptor);
+        }
+
+        await _graphApi.PasteCellsAsync(activityNodes, []);
+    }
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Studio.Workflows.Designer/Components/SequenceFlowDesigner.razor.cs:119]: Fixed in 66b71f66. Extracted shared pasted-activity identity regeneration into `RegenerateActivityIdentity`, making the top-level and embedded recursion path explicit.

File: src/modules/Elsa.Studio.Workflows.Designer/Services/SequenceMapper.cs [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+    public static void SetLayoutOrientation(JsonObject sequence, string? orientation)
+    {
+        var normalized = IsHorizontal(orientation) ? HorizontalOrientation : VerticalOrientation;
+        var metadata = sequence.TryGetPropertyValue("metadata", out var metadataNode) && metadataNode is JsonObject existingMetadata
+            ? existingMetadata
+            : new JsonObject();
+
+        metadata[LayoutOrientationMetadataKey] = normalized;
+        sequence["metadata"] = metadata;
+    }
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Studio.Workflows.Designer/Services/SequenceMapper.cs:110]: Fixed in 66b71f66 with tests. Vertical/default orientation no longer writes metadata for legacy sequences, and returning to vertical clears the orientation key.

File: src/modules/Elsa.Studio.Workflows.Designer/ClientLib/src/react-designer/components/Designer.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Studio.Workflows.Designer/ClientLib/src/react-designer/components/Designer.tsx:1286]: No code change in this PR. Same rationale as the earlier localization thread: the current React designer needs a shared localization bridge for client-only strings; adding a Sequence-only mechanism would leave the broader React surface inconsistent.

File: src/modules/Elsa.Studio.Workflows.Designer/Components/SequenceFlowDesigner.razor.cs [hunk 10]
```diff
@@ -0,0 +1,328 @@
...
+        {
+            _sequence = Sequence;
+            await _rateLimitedLoadSequenceAction.InvokeAsync();
+        }
+
+        if (!Equals(_activityStats, ActivityStats))
+        {
+            _activityStats = ActivityStats;
+            await _rateLimitedLoadSequenceAction.InvokeAsync();
+        }
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Studio.Workflows.Designer/Components/SequenceFlowDesigner.razor.cs:262]: Fixed in 66b71f66. `OnParametersSetAsync` now records whether either parameter changed and invokes the debounced reload once per parameter-set cycle.

File: src/modules/Elsa.Studio.Workflows.Designer/Components/SequenceFlowDesigner.razor.cs [hunk 11]
```diff
@@ -0,0 +1,338 @@
...
+    }
+
+    public async Task<JsonObject> ReadSequenceAsync()
+    {
+        if (_graphApi is null) return Sequence;
+
+        var serializerOptions = new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
+        var data = await _graphApi.ReadGraphAsync();
+        var cells = data.GetProperty("cells").EnumerateArray();
+        var nodes = cells.Where(x => x.GetProperty("shape").GetString() == "elsa-activity")
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Studio.Workflows.Designer/Components/SequenceFlowDesigner.razor.cs]: Fixed in 1a549c7e. `ReadSequenceAsync` now uses `TryGetProperty("shape", out var shape)` before comparing the shape value, so unexpected cells without `shape` are ignored instead of throwing.

File: src/modules/Elsa.Studio.Workflows/DiagramDesigners/Sequences/SequenceDesignerWrapper.razor.cs [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+
+        if (ActivitySelected.HasDelegate)
+            await ActivitySelected.InvokeAsync(newActivity);
+    }
+
+    private void OnDragOver(DragEventArgs e)
+    {
+        if (DragDropManager.Payload is not ActivityDescriptor)
+        {
+            e.DataTransfer.DropEffect = "none";
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Studio.Workflows/DiagramDesigners/Sequences/SequenceDesignerWrapper.razor.cs]: Fixed in 1a549c7e. Extracted shared Sequence child activity construction into `SequenceActivityFactory.CreateActivity`; both `SequenceDesignerWrapper` and `SequenceFlowDesigner` now use it.

File: src/modules/Elsa.Studio.Workflows/DiagramDesigners/Sequences/SequenceDesignerWrapper.razor.cs [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+/// <summary>
+/// A wrapper around the Sequence designer component.
+/// </summary>
+public partial class SequenceDesignerWrapper
+{
+    [Parameter] public JsonObject Sequence { get; set; } = null!;
+    [Parameter] public IDictionary<string, ActivityStats>? ActivityStats { get; set; }
+    [Parameter] public bool IsReadOnly { get; set; }
+    [Parameter] public EventCallback<JsonObject> ActivitySelected { get; set; }
+    [Parameter] public EventCallback<JsonObject> ActivityUpdated { get; set; }
```
[the remark this answers is not in the document]
Comment (sfmskywalker) [reply src/modules/Elsa.Studio.Workflows/DiagramDesigners/Sequences/SequenceDesignerWrapper.razor.cs:25]: Fixed in 0d9c133a. `SequenceDesignerWrapper` now forwards `ActivityUpdated` into `SequenceFlowDesigner`, and the Sequence designer raises it after activity updates/adds so the host can observe dirty-state changes.

**Comment by sfmskywalker** [comment]
Replying to the Greptile summary from https://github.com/elsa-workflows/elsa-studio/pull/819#issuecomment-4471375575: the blocking Earlier/Later reorder issue and the robustness/duplication follow-ups were fixed in 1a549c7e, with validation passing via `dotnet test src/modules/Elsa.Studio.Workflows.Designer.Tests/Elsa.Studio.Workflows.Designer.Tests.csproj --no-restore` and `npm run build` in ClientLib.
